Tests/TuistDependenciesTests/SwiftPackageManager/SwiftPackageManagerInteractorTests.swift

```diff
@@ -28,49 +33,103 @@ final class SwiftPackageManagerInteractorTests: TuistUnitTestCase {
         let rootPath = try TemporaryDirectory(removeTreeOnDeinit: true).path
         let dependenciesDirectory = rootPath
             .appending(component: Constants.DependenciesDirectory.name)
+        let lockfilesDirectory = dependenciesDirectory
+            .appending(component: Constants.DependenciesDirectory.lockfilesDirectoryName)
+        let swiftPackageManagerDirectory = dependenciesDirectory
+            .appending(component: Constants.DependenciesDirectory.swiftPackageManagerDirectoryName)
 
-        try createFiles([
-            "Package.resolved",
-            ".build/manifest.db",
-            ".build/workspace-state.json",
-            ".build/artifacts/foo.txt",
-            ".build/checkouts/Alamofire/Info.plist",
-            ".build/repositories/checkouts-state.json",
-            ".build/repositories/Alamofire-e8f130fe/config",
+        let depedencies = SwiftPackageManagerDependencies([
+            .remote(url: "https://github.com/Alamofire/Alamofire.git", requirement: .upToNextMajor("5.2.0")),
         ])
 
-        let platforms = Set<Platform>([.iOS, .watchOS, .macOS, .tvOS])
-        let command = ["swift", "package", "--package-path", "\(try temporaryPath().pathString)", "resolve"]
-        system.succeedCommand(command)
-        system.swiftVersionStub = { "5.3" }
+        swiftPackageManager.resolveStub = { [unowned self] path in
```

Comment:
you can just use `self` directly, we don't care about reference cycle since we nullify the dependencies in `tearDown`